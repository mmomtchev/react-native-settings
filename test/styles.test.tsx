import React from 'react';
import {Text, StyleSheet} from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';

import Settings, {SettingsElement, SettingsStyle} from '@mmomtchev/react-native-settings';
import {waitForSpinner} from './data';

describe('styles', () => {
    const settings: SettingsElement[] = [
        {
            label: 'Name',
            type: 'string',
            display: (s) => (s && s.length ? s : 'empty'),
            get: () => '',
            set: (v) => undefined
        },
        {
            label: 'Intelligence',
            type: 'enum',
            values: ['A'],
            display: () => 'Alpha',
            get: () => 'A',
            set: (v) => undefined
        },
        {
            label: 'Header',
            type: 'section',
            elements: [
                {
                    label: 'Wings',
                    type: 'boolean',
                    get: () => true,
                    set: (v) => undefined
                }
            ]
        }
    ];

    const styles: SettingsStyle = StyleSheet.create({
        screen: {
            flex: 1,
            padding: 4
        },
        list: {
            padding: 6
        },
        header: {
            backgroundColor: '#dce3f4',
            height: 42,
            flexDirection: 'row',
            alignItems: 'center',
            paddingStart: 12,
            paddingEnd: 12
        },
        item: {
            fontFamily: 'sans-serif',
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
            height: 41,
            borderColor: '#c8c7cc',
            borderBottomWidth: 2,
            paddingStart: 10,
            paddingEnd: 10
        },
        input: {
            fontFamily: 'serif'
        },
        label: {
            color: 'blue'
        },
        value: {
            color: 'red'
        },
        spinner: {
            position: 'absolute',
            left: 25,
            right: 0,
            top: 0,
            bottom: 0,
            alignItems: 'center',
            justifyContent: 'center',
            zIndex: 10
        }
    });

    it('render', async () => {
        const r = render(
            <NavigationContainer>
                <Settings styles={styles} settings={settings} />
            </NavigationContainer>
        );
        await waitForSpinner(r);
        expect(r.toJSON()).toMatchSnapshot();
        r.unmount();
    });
});

describe('components', () => {
    const settings: SettingsElement[] = [
        {
            label: 'Name',
            type: 'string',
            display: (s) => (s && s.length ? s : 'empty'),
            get: () => '',
            set: (v) => undefined,
            jsxLabel: <Text>_N_A_M_E</Text>
        },
        {
            label: 'Intelligence',
            type: 'enum',
            values: ['A'],
            display: () => 'Alpha',
            get: () => 'A',
            set: (v) => undefined,
            jsxLabel: <Text>1nt3ll1g3nc3</Text>
        },
        {
            label: 'Header',
            type: 'section',
            jsxLabel: <Text>FEATURES</Text>,
            elements: [
                {
                    label: 'Wings',
                    type: 'boolean',
                    get: () => true,
                    set: (v) => undefined,
                    jsxLabel: <Text>wiiings</Text>
                }
            ]
        }
    ];

    it('render', async () => {
        const r = render(
            <NavigationContainer>
                <Settings settings={settings} />
            </NavigationContainer>
        );
        await waitForSpinner(r);
        expect(r.toJSON()).toMatchSnapshot();
        r.unmount();
    });
});
