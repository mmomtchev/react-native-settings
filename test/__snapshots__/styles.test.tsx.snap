// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components render 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 5,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreenStack
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onGestureCancel={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          stackPresentation="push"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backTitleVisible={true}
            backgroundColor="rgb(255, 255, 255)"
            color="rgb(0, 122, 255)"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={true}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="ReactNativeSettingsMain"
            titleColor="rgb(28, 28, 30)"
            topInsetEnabled={false}
            translucent={false}
          />
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "padding": 5,
                  }
                }
              >
                <View
                  accessibilityState={
                    Object {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    Object {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "borderBottomWidth": 2,
                        "borderColor": "#c8c7cc",
                        "flexDirection": "row",
                        "height": 40,
                        "justifyContent": "space-between",
                        "paddingEnd": 10,
                        "paddingStart": 10,
                      }
                    }
                  >
                    <Text>
                      _N_A_M_E
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#c8c7cc",
                        }
                      }
                    >
                      empty
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityState={
                    Object {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    Object {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "borderBottomWidth": 2,
                        "borderColor": "#c8c7cc",
                        "flexDirection": "row",
                        "height": 40,
                        "justifyContent": "space-between",
                        "paddingEnd": 10,
                        "paddingStart": 10,
                      }
                    }
                  >
                    <Text>
                      1nt3ll1g3nc3
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#c8c7cc",
                        }
                      }
                    >
                      Alpha
                    </Text>
                  </View>
                </View>
                <View>
                  <Text>
                    FEATURES
                  </Text>
                  <View
                    style={
                      Object {
                        "padding": 5,
                      }
                    }
                  >
                    <View
                      accessibilityState={
                        Object {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        Object {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "borderBottomWidth": 2,
                            "borderColor": "#c8c7cc",
                            "flexDirection": "row",
                            "height": 40,
                            "justifyContent": "space-between",
                            "paddingEnd": 10,
                            "paddingStart": 10,
                          }
                        }
                      >
                        <Text>
                          wiiings
                        </Text>
                        <View
                          pointerEvents="none"
                        >
                          <RCTSwitch
                            accessibilityRole="switch"
                            onChange={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "height": 31,
                                "width": 51,
                              }
                            }
                            value={true}
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenStack>
    </RNCSafeAreaProvider>
  </View>
</View>
`;

exports[`styles render 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 4,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreenStack
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onGestureCancel={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          stackPresentation="push"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backTitleVisible={true}
            backgroundColor="rgb(255, 255, 255)"
            color="rgb(0, 122, 255)"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={true}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="ReactNativeSettingsMain"
            titleColor="rgb(28, 28, 30)"
            topInsetEnabled={false}
            translucent={false}
          />
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "padding": 6,
                  }
                }
              >
                <View
                  accessibilityState={
                    Object {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    Object {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "borderBottomWidth": 2,
                        "borderColor": "#c8c7cc",
                        "flexDirection": "row",
                        "fontFamily": "sans-serif",
                        "height": 41,
                        "justifyContent": "space-between",
                        "paddingEnd": 10,
                        "paddingStart": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "blue",
                        }
                      }
                    >
                      Name
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "red",
                        }
                      }
                    >
                      empty
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityState={
                    Object {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    Object {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "borderBottomWidth": 2,
                        "borderColor": "#c8c7cc",
                        "flexDirection": "row",
                        "fontFamily": "sans-serif",
                        "height": 41,
                        "justifyContent": "space-between",
                        "paddingEnd": 10,
                        "paddingStart": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "blue",
                        }
                      }
                    >
                      Intelligence
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "red",
                        }
                      }
                    >
                      Alpha
                    </Text>
                  </View>
                </View>
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#dce3f4",
                        "flexDirection": "row",
                        "height": 42,
                        "paddingEnd": 12,
                        "paddingStart": 12,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "blue",
                        }
                      }
                    >
                      Header
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "padding": 6,
                      }
                    }
                  >
                    <View
                      accessibilityState={
                        Object {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        Object {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "borderBottomWidth": 2,
                            "borderColor": "#c8c7cc",
                            "flexDirection": "row",
                            "fontFamily": "sans-serif",
                            "height": 41,
                            "justifyContent": "space-between",
                            "paddingEnd": 10,
                            "paddingStart": 10,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "blue",
                            }
                          }
                        >
                          Wings
                        </Text>
                        <View
                          pointerEvents="none"
                        >
                          <RCTSwitch
                            accessibilityRole="switch"
                            onChange={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "height": 31,
                                "width": 51,
                              }
                            }
                            value={true}
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenStack>
    </RNCSafeAreaProvider>
  </View>
</View>
`;
